react/week4/user-search/src/SearchList.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useContext } from "react";
+import { SearchContext } from "./SearchContext";
+
+//const github_api = "https://api.github.com/search/users?q=";
+
+function SearchList() {
+    const searchingContext = useContext(SearchContext);
+    const allSearcher = () => {
+        if (searchingContext.input !== ""){
+            searchingContext.setLoading(true);
+            fetch(`https://api.github.com/search/users?q=${searchingContext.input}`)
+            .then((response) => response.json())
+            .then((data)=>{
+                searchingContext.setLoading(false);
+                if(data.search) {
+                    searchingContext.setSearch(data.search);
+
+                }
+                else if (data.count === 0) {
+                 searchingContext.setSearch("not found..");   
+                }
+                else{
+                    searchingContext.setUser(data.items.map((item)=>item.login));
+                }
```

Comment:
You're not handling errors from the api, so this will fail when the api fails. 
You should use the `data.message` to display an error message.